extensions/amp-byside-placeholder/0.1/amp-byside-placeholder.js

```diff
@@ -0,0 +1,348 @@
+      this.element.setAttribute('frameborder', '0');
+    }
+  }
+
+  /** @override */
+  createPlaceholderCallback() {
+    const placeholder = this.win.document.createElement('div');
+    placeholder.setAttribute('placeholder', '');
+    const image = this.win.document.createElement('amp-img');
+    image.setAttribute('noprerender', '');
+    image.setAttribute('layout', 'fill');
+    image.setAttribute('referrerpolicy', 'origin');
+
+    this.propagateAttributes(['alt'], image);
+
+    // Use custom placeholder or use default transparent placeholder image
+    const placeholderImg = this.element.getAttribute('placeholder') ||
+      this.baseUrl_ + 'images/placeholder.png';
+    image.setAttribute('src', placeholderImg);
+
+    placeholder.appendChild(image);
+    return placeholder;
+  }
+
+  /** @override */
+  layoutCallback() {
+    const iframe = this.element.ownerDocument.createElement('iframe');
+    this.iframe_ = iframe;
+
+    const width = this.element.getAttribute('width') || DEFAULT_IFRAME_WIDTH_;
+    const height = this.element.getAttribute('height') ||
+        DEFAULT_IFRAME_HEIGHT_;
+
+    iframe.name = 'amp_byside_placeholder_iframe' + iframeCount_++;
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowtransparency', 'true');
+    iframe.setAttribute('title', this.element.getAttribute('alt') || '');
+
+    if (this.element.hasAttribute('width')) {
+      iframe.setAttribute('width', width);
+    }
+    if (this.element.hasAttribute('height')) {
+      iframe.setAttribute('height', height);
+    }
+
+    setStyles(iframe, {
+      'opacity': 0,
+    });
+
+    const self = this;
+
+    return this.generateSrcUrl_().then(src => {
+      this.iframeSrc_ = src;
+      iframe.src = this.iframeSrc_;
+
+      listenFor(iframe, 'embed-size', data => {
+        this.updateSize_(data['height'], data['width']);
+      });
+
+      this.applyFillContent(iframe);
+      this.element.appendChild(iframe);
+    }).then(() => {
+      this.iframePromise_ = this.loadPromise(iframe).then(() => {
+        this.getVsync().mutate(() => {
+          setStyles(iframe, {
+            'opacity': 1,
+            'width': width + 'px',
+            'height': height + 'px',
+          });
+        });
+      });
+    }).then(() => self);
+  }
+
+  /** @private */
+  generateOrigin_() {
+    const subDomain = this.webcareZone_ === MAIN_WEBCARE_ZONE_ ?
+      MAIN_WEBCARE_ZONE_SUBDOMAIN_ :
+      this.webcareZone_;
+    return 'https://' + subDomain + '.byside.com';
+  }
+
+  /** @private */
+  generateSrcUrl_() {
+    const src = this.baseUrl_ + 'placeholder.php';
+    const params = dict();
+
+    params['label'] = this.label_;
+    params['webcare_id'] = this.webcareId_;
+    params['bwch'] = this.channel_ || '';
+    params['lang'] = this.lang_ || '';
+    params['fid'] = this.fid_ || '';
+    params['bwit'] = (this.fid_ ? 'I' : 'A');
+    params['tuid'] = 'CLIENT_ID(byside_webcare_tuid)';
```

Comment:
Can't send this as it currently stands. If this is for analytics purposes please follow https://github.com/ampproject/amphtml/blob/f8e32a528f0fb02dcec54f58d7d8f2320db55673/extensions/amp-analytics/amp-components-analytics.md guidelines. @zhouyx can help if any questions. Please note the `Documentation requirements` as well.